src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -40,6 +41,7 @@ public async Task<string> GetSecretValue(KeyVaultSecretIdentifier secretIdentifi
             string secretValue = null;
 
             if (_cachedKeyVaultSecrets.TryGetValue(key, out CachedKeyVaultSecret cachedSecret) &&
+                    (cachedSecret.SourceId == secretIdentifier.SourceId) &&
                     (!cachedSecret.RefreshAt.HasValue || DateTimeOffset.UtcNow < cachedSecret.RefreshAt.Value))
```

Comment:
It looks like we need to cache by source id. By the way can you give me an example of what that might look like?

Reply:
Source id looks like: https://{vault name}.vault.azure.net/secrets/{secret name}

Reply:
Do you mean we use source id + AppConfig key as cache key?

Reply:
> Source id looks like: https://{vault name}.vault.azure.net/secrets/{secret name}

It should include the version too, right?

Reply:
Discussed offline w/ Avani, based on our usage of the secret provider caching by key works as expected. Resolving.

Reply:
One thing I ran into, I'm not sure how that would work for calling `InvalidateCache(ConfigurationSetting)` from `RefreshAsync`